src/governance/LendingTermOnboarding.sol

```diff
@@ -0,0 +1,208 @@
+    address public immutable lendingTermImplementation;
+    /// @notice immutable reference to the guild token
+    address public immutable guildToken;
+    /// @notice immutable reference to the gauge type to use for the terms onboarded
+    uint256 public immutable gaugeType;
+
+    /// @notice timestamp of creation of a term
+    /// (used to check that a term has been created by this factory)
+    mapping(address=>uint256) public created;
+
+    /// @notice reference to profitManager to set in created lending terms
+    address public immutable profitManager;
+    /// @notice reference to auctionHouse to set in created lending terms
+    address public immutable auctionHouse;
+    /// @notice reference to creditMinter to set in created lending terms
+    address public immutable creditMinter;
+    /// @notice reference to creditToken to set in created lending terms
+    address public immutable creditToken;
+
+    constructor(
+        address _lendingTermImplementation,
+        LendingTerm.LendingTermReferences memory _lendingTermReferences,
+        uint256 _gaugeType,
+        address _core,
+        address _timelock,
+        uint256 initialVotingDelay,
+        uint256 initialVotingPeriod,
+        uint256 initialProposalThreshold,
+        uint256 initialQuorum
+    )
+        VoltGovernor(
+            _core,
+            _timelock,
+            _lendingTermReferences.guildToken,
+            initialVotingDelay,
+            initialVotingPeriod,
+            initialProposalThreshold,
+            initialQuorum
+        )
+    {
+        lendingTermImplementation = _lendingTermImplementation;
+        guildToken = _lendingTermReferences.guildToken;
+        gaugeType = _gaugeType;
+        profitManager = _lendingTermReferences.profitManager;
+        auctionHouse = _lendingTermReferences.auctionHouse;
+        creditMinter = _lendingTermReferences.creditMinter;
+        creditToken = _lendingTermReferences.creditToken;
+    }
+
+    /// @notice Create a new LendingTerm and initialize it.
+    function createTerm(LendingTerm.LendingTermParams calldata params) external returns (address) {
+        // must be an ERC20 (maybe, at least it prevents dumb input mistakes)
+        (bool success, bytes memory returned) = params.collateralToken.call(abi.encodeWithSelector(IERC20.totalSupply.selector));
+        require(success && returned.length == 32, "LendingTermOnboarding: invalid collateralToken");
+    
+        require(
+            params.maxDebtPerCollateralToken != 0, // must be able to mint non-zero debt
+            "LendingTermOnboarding: invalid maxDebtPerCollateralToken"
+        );
+
+        require(
+            params.interestRate < 1e18, // interest rate [0, 100[% APR
+            "LendingTermOnboarding: invalid interestRate"
+        );
+
+        require(
+            params.maxDelayBetweenPartialRepay < 31557601, // periodic payment every [0, 1 year]
```

Comment:
use constant 1 years instead of magic number

Reply:
`years` is a deprecated time unit & I took `31557600` as a fixed value in `LendingTerm`